### PR TITLE
enhancing ltp_fsstress and fixing pre_clean issue

### PR DESCRIPTION
**Pull request by Naresh-ibm**
fixing the pre_cleanup issue and did some enhancement
to the existing script for better usage of script

Signed-off-by: Naresh Bannoth <nbannoth@in.ibm.com>

### CODE REVIEW

File: io/disk/ltp_fsstress.py [hunk 1]
```diff
@@ -139,38 +173,166 @@ def create_fs(self, l_disk, mountpoint, fstype):
...
+            process.system(cmd, shell=True, ignore_status=True)
+            if disk.fs_exists(l_disk):
+                return False
+            return True
+
+        def is_disk_unmounted():
+            cmd = "umount %s" % l_disk
+            cmd1 = 'umount /dev/mapper/avocado_vg-avocado_lv'
+            process.system(cmd, shell=True, ignore_status=True)
+            process.system(cmd1, shell=True, ignore_status=True)
```
Comment (abdhaleegit) [inline io/disk/ltp_fsstress.py:269]: Why duplicate commands ? cant we pass disk value and a single process.system(cmd, can take care of both ?

Comment (abdhaleegit) [reply io/disk/ltp_fsstress.py:269]: param initialization should be outside the function always

File: io/disk/ltp_fsstress.py [hunk 2]
```diff
@@ -139,38 +173,166 @@ def create_fs(self, l_disk, mountpoint, fstype):
...
+                return False
+            return True
+
+        self.log.info("checking if disk is mounted.")
+        if disk.is_disk_mounted(l_disk):
+            self.log.info("%s is mounted, unmounting it ....", l_disk)
+            if wait.wait_for(is_disk_unmounted, timeout=10):
+                self.log.info("%s unmounted successfully" % l_disk)
+            else:
+                self.err_mesg.append("%s unmount failed", l_disk)
```
Comment (abdhaleegit) [inline io/disk/ltp_fsstress.py:287]: log only if fails.  too much logging is not needed

File: io/disk/ltp_fsstress.py.data/ltp_fsstress.yaml [hunk 3]
```diff
@@ -1,7 +1,8 @@
-disk:
-script: 'fsstress'
-dir: '/mnt/nfs'
-fsstress_loop: 1
+disk: ''
+dir: ''
+fsstress_loop:
+nval: '250'
+pval: '250'
```
Comment (abdhaleegit) [inline io/disk/ltp_fsstress.py.data/ltp_fsstress.yaml:5]: Does readme explains the values to be given fsstress_loop:
nval: '250'
pval: '250'

**Review by abdhaleegit — CHANGES_REQUESTED** [review]
@Naresh-ibm can we avoid too much loginfo.. I would suggested to not log if things pass.. log only when things fail.. this applies all cleanup code PRs

File: io/disk/ltp_fsstress.py (hunk 1 above)
Comment (Naresh-ibm) [reply io/disk/ltp_fsstress.py:269]: will put them outside of the method.

File: io/disk/ltp_fsstress.py.data/ltp_fsstress.yaml (hunk 3 above)
Comment (Naresh-ibm) [reply io/disk/ltp_fsstress.py.data/ltp_fsstress.yaml:5]: we do not have the Readme file. will add one with all required info

**Pull request closed without merging**